### ISSUE TITLE
optipng formula fails

### ISSUE THREAD

**Issue by black-snow** [Linuxbrew/homebrew-core#8442]
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single formula?
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
https://gist.github.com/a3f6b2f4fb8f50fb3657ee4f335a5842

> What you were trying to do (and why)

I tried to `brew install optipng`.

- What happened (include command output)

```
$ brew install optipng
nice: cannot set niceness: Permission denied
==> Downloading https://downloads.sourceforge.net/project/optipng/OptiPNG/optipng-0.7.7/optipng-0.7.7.tar.gz
Already downloaded: /home/rbergmann/.cache/Homebrew/optipng-0.7.7.tar.gz
==> ./configure --with-system-zlib --prefix=/home/linuxbrew/.linuxbrew/Cellar/optipng/0.7.7 --mandir=/home/linuxbrew/.linuxbrew/Cellar/optipng/0.7.7/share/man
==> make install
Last 15 lines from /home/rbergmann/.cache/Homebrew/Logs/optipng/02.make:
gcc-5 -c -I../zlib  -O2 -Wall -Wextra -o pngwutil.o pngwutil.c
gcc-5 -c -I../zlib  -O2 -Wall -Wextra -o pngtest.o pngtest.c
ar rcs libpng.a png.o pngerror.o pngget.o pngmem.o pngpread.o pngread.o pngrio.o pngrtran.o pngrutil.o pngset.o pngtrans.o pngwio.o pngwrite.o pngwtran.o pngwutil.o
ranlib libpng.a
gcc-5  -L../zlib -o pngtest pngtest.o libpng.a -lz -lm
/home/linuxbrew/.linuxbrew/opt/binutils/bin/ld: cannot find -lz
collect2: error: ld returned 1 exit status
Makefile:66: recipe for target 'pngtest' failed
make[2]: *** [pngtest] Error 1
make[2]: Leaving directory '/tmp/optipng-20180725-10486-c5ithd/optipng-0.7.7/src/libpng'
Makefile:129: recipe for target '../libpng/libpng.a' failed
make[1]: *** [../libpng/libpng.a] Error 2
make[1]: Leaving directory '/tmp/optipng-20180725-10486-c5ithd/optipng-0.7.7/src/optipng'
Makefile:14: recipe for target 'install' failed
make: *** [install] Error 2

READ THIS: https://github.com/Linuxbrew/brew/wiki/troubleshooting
Please do not report this issue to Homebrew/brew or Homebrew/core, which support macOS only.
```

> What you expected to happen

A clean install.

**Comment by iMichka** [comment]
Hi. I think you just need to add: `depends_on "zlib"` to the formula to make this work. Would you like to open a pull request for that?

**Comment by black-snow** [comment]
There you go: https://github.com/Linuxbrew/homebrew-core/pull/8761

Doesn't this somewhat collide with `--with-system-zlib`? Sure, if there ain't no zlib on the system we're effed but if there is one we will install another unused one into the cellar, right?

**Comment by iMichka** [comment]
Fixed by #8761
